src/checkers/flymake-collection-clang-tidy.el

```diff
@@ -0,0 +1,92 @@
+;;; flymake-collection-clang-tidy.el --- Clang-tidy diagnostic function -*- lexical-binding: t -*-
+
+;; Copyright (c) 2024 Abdelhak Bougouffa
+
+;; Permission is hereby granted, free of charge, to any person obtaining a copy
+;; of this software and associated documentation files (the "Software"), to deal
+;; in the Software without restriction, including without limitation the rights
+;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+;; copies of the Software, and to permit persons to whom the Software is
+;; furnished to do so, subject to the following conditions:
+
+;; The above copyright notice and this permission notice shall be included in all
+;; copies or substantial portions of the Software.
+
+;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+;; SOFTWARE.
+
+;;; Commentary:
+
+;; `flymake' static checker for C/C++ using "clang-tidy".
+;; Inspired by `flycheck-clang-tidy' - https://github.com/ch1bo/flycheck-clang-tidy
+
+;;; Code:
+
+(require 'flymake)
+(require 'flymake-collection)
+
+(eval-when-compile
+  (require 'flymake-collection-define))
+
+;;; Custom variables
+
+(defcustom flymake-collection-clang-tidy-build-path "build"
+  "Clang build directory."
+  :type '(choice (const nil) directory)
+  :group 'minemacs-prog)
```

Comment:
These group variables should probably be moved to a new group in flymake collection. I'm guessing this is what you had in your own .emacs :thinking:.

Reply:
Yep, it comes from my own config! It is fixed now!